### PR TITLE
265: Add fragments director; begin fragments testing

### LINKED ISSUE #265: Test Fragments
- [x] Write unit tests for all fragments in `app/fragments`
- [x] Document all fragments with JSDoc-ish @pichot 

These objects are all simple `EmberObject`s. They nest within in each other, meaning the input of one fragment is often an array of other fragments.

The nesting pattern is as follows:

`Building` which is the most basic input of a school building, identified by level, district, and subdistrict.
`SchoolTotals` consumes a list of all `Building`s, level, district, and subdistrict
`AggregateTotals` consumes a list of `SchoolTotals` by level
`SchoolLevelTotals` consumes a list of `AggregateTotals` by level

Given this structure, there is definitely need for better naming of these fragments and potentially some refactoring.

### PR DESCRIPTION
**Pull request by trbmcginnis**
@pichot , if you could add notes about the changes you made here.

- Added tests for building, aggregate totals, school totals, and school level totals fragment calculations

Closes #265

### CODE REVIEW

File: frontend/tests/unit/fragments/public-schools/school-level-totals-test.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+      capacityWithActionTotal: 2,
+      utilizationNoActionTotal: 3
+    }));
+
+    // seatsToMitigateUtilization = enrollWithActionTotal - capacityWithActionTotal - 1
+    // seatsToMitigateUtilization = 7 - 2 - 1 = 4
+    // seatsToMitigateChange = (enrollWithActionTotal / (utilizationNoActionTotal + 0.0499)) - capacityWithActionTotal
+    // seatsToMitigateChange = (7 / (3 + 0.0499)) - 2 = 0.295157
+    // Math.ceil(seatsToMitigateChange) = 1
+    // seatsToMitigateUtilization > seatsToMitigateChange, SO RETURN seatsToMitigateChange,
```
Comment (allthesignals) [inline frontend/tests/unit/fragments/public-schools/school-level-totals-test.js]: Should this be deleted? Not sure it adds clarity

File: frontend/tests/unit/fragments/public-schools/school-totals-test.js [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+      level: 'hs'
+    }));
+
+    assert.deepEqual(hs_school_totals.buildings, [
+      Building.create({
+        level: 'hs',
+        district: 1,
+        subdistrict: 2
+      })
+    ]);
```
Comment (allthesignals) [inline frontend/tests/unit/fragments/public-schools/school-totals-test.js]: So is the behavior we're testing here that it `.buildings` returns a certain filtered subset? Only `hs`?

File: frontend/tests/unit/fragments/public-schools/school-level-totals-test.js (hunk 1 above)
Comment (trbmcginnis) [reply frontend/tests/unit/fragments/public-schools/school-level-totals-test.js]: I added this for the tester's purpose. It helped me work through what the end values should be and I feel like it'll help future testers. Also could just save a little time for folks looking at the code in the future. I'd love to keep it there

File: frontend/tests/unit/fragments/public-schools/aggregate-totals-test.js [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+import { run } from '@ember/runloop';
+
+import AggregateTotals from 'labs-ceqr/fragments/public-schools/AggregateTotals';
+import SchoolTotals from 'labs-ceqr/fragments/public-schools/SchoolTotals';
+
+module('Unit | Fragment | AggregateTotals', function(hooks) {
+  setupTest(hooks);
+
+  test('#enrollExistingConditions is calculated correctly', function(assert) {
+    let school_totals = run(() => SchoolTotals.create({
```
[the remark this answers is not in the document]
Comment (trbmcginnis) [reply frontend/tests/unit/fragments/public-schools/aggregate-totals-test.js]: I'm not totally sure, @pichot do you know the exact reasoning?

**Comment by trbmcginnis** [comment]
> _Excellent_, thanks for owning this.
> 
> Just some thoughts:
> 
> 1. Why do all the object `create` methods need to be wrapped in a `run`? If there's something async we should convert the test `function`s to async/await. Otherwise we can remove the `run` wrappers.
> 2. How were the values for the assertions determined? For example, was `assert.equal(school_level_totals.subdistricts.enrollTotal, 400)` based on the output of the app now or verified independently? J/w.

1. @pichot Do you know the exact reasoning for using run?
2. I just put in random values to test the behavior. If we need more realistic values, I can go in and edit them to reflect more likely situations.

File: frontend/tests/unit/fragments/public-schools/school-totals-test.js (hunk 2 above)
Comment (trbmcginnis) [reply frontend/tests/unit/fragments/public-schools/school-totals-test.js]: Yes that's what we're testing. I can add comments above the test to explain that

File: frontend/tests/unit/fragments/public-schools/school-level-totals-test.js (hunk 1 above)
Comment (allthesignals) [reply frontend/tests/unit/fragments/public-schools/school-level-totals-test.js]: Sure, whatever helps - I just thought it was commented-out code.

Comment (trbmcginnis) [reply frontend/tests/unit/fragments/public-schools/school-level-totals-test.js]: ah gotcha!

**Pull request merged**